image/image.go

```diff
@@ -157,29 +208,40 @@ func sanitize(s string) string {
 // NOTE: The manifest may have been modified in the process; DO NOT reserialize and store the return value
 // if you want to preserve the original manifest; use the blob returned by Manifest() directly.
 // NOTE: It is essential for signature verification that the object is computed from the same manifest which is returned by Manifest().
-func (i *genericImage) getSchema1Manifest() (genericManifest, error) {
+func (i *genericImage) getParsedManifest() (genericManifest, error) {
 	manblob, err := i.Manifest()
 	if err != nil {
 		return nil, err
 	}
-	mschema1 := &manifestSchema1{}
-	if err := json.Unmarshal(manblob, mschema1); err != nil {
-		return nil, err
-	}
-	if err := fixManifestLayers(mschema1); err != nil {
-		return nil, err
+	mt := manifest.GuessMIMEType(manblob)
```

Comment:
`i.Manifest` may have known the MIME type; we should use that, if known, and only guess if unknown.  (Perhaps an `i.manifestAndMIMEType` doing the caching, and `i.Manifest` a public method which calls `i.manifestAndMIMEType` and discards the MIME type? Or do we want to make the MIME type public for conversions anyway?)